meter-health/meter_health.py

```diff
@@ -1,98 +0,0 @@
-import requests
-import csv
-import logging
-logging.basicConfig(filename='example.log',level=logging.DEBUG)
-from oauthlib.oauth2 import LegacyApplicationClient
-from requests_oauthlib import OAuth2Session
-
-class MeterHealth:
-    # Change the values here to match your user and your API credentials
-    CLIENT_ID = 'WRITE YOUR API CLIENT ID HERE'
-    CLIENT_SECRET = 'WRITE YOUR API CLIENT SECRET HERE'
-    USERNAME = 'WRITE YOUR USERNAME HERE'
-    PASSWORD = 'WRITE YOUR PASSWORD HERE'
-    SCOPE = ['admin_organization']
-
-    # URL information for the API. ONLY change this method if you know what you are doing
-    BASE_URL = "http://54.90.173.204:8080"
-    ORGANIZATIONS ="/api/v2.json"
-    METERS = "/api/v2/organizations/%d/meters.json"
-    TOKEN_URL = "/oauth/token"
-
-    def __init__(self):
-        self.token = self.retrieve_token()
-        self.payload = {'access_token': self.token}
-
-    # create_report: is the main method that calls
-    # all that need to be done:
-    # 1. retrieve organizations
-    # 2. iterate over the organizations to get meters
-    # 3. create the csv file
-    def create_report(self):
-        if self.token:
-            organizations = self.retrieve_organizations()
-            if organizations:
-                total_org = len(organizations)
-                organization_index = 1
-                print("We found %d organizations \n" % total_org)
-                with open('example_file_test.csv', 'w+') as csv_file:
-                    fieldnames = ["Organization Name", "Meter ID", "Meter Name",
-                                    "Meter Type", "Meter Status", "Last Reading"]
-                    writer = csv.writer(csv_file)
-                    writer.writerow(fieldnames)
-                    for organization in organizations:
-                        print("Retrieving meters for organization %d of %d ...\n" % (organization_index, total_org))
-                        organization_index += 1
-                        meter_info = self.retrieve_meter_info_for(organization['remote_id'])
-                        for meter in meter_info:
```

Comment:
I think that you can replace the for loop with a map() function, it's faster than a "for".
[https://docs.python.org/2/library/functions.html#map](url)